src/client/datascience/interactive-window/nativeInteractiveWindow.ts

```diff
@@ -376,7 +379,21 @@ export class NativeInteractiveWindow implements IInteractiveWindowLoadable {
             }
 
             if (isDebug) {
-                await this.jupyterDebugger.startDebugging(notebook);
+                const interpreter = notebook.getKernelConnection()?.interpreter;
+                let execService: IPythonExecutionService | undefined;
+                if (interpreter) {
+                    execService = await this.pythonExecFactory.createActivatedEnvironment({
+                        interpreter: notebook.getKernelConnection()?.interpreter
+                    });
+                }
+                let ipykernelVersion: string | undefined;
+                if (execService) {
+                    const result = await execService
```

Comment:
Can we use the `isProductVersionCompatible` Python extension API function instead of reimplementing here? IIRC David added it to the API?

Reply:
No we can't, its not a compatibility check, we need the version information.

Reply:
My plan is to enable caching as well (separate pr), also not sure we need to move stuff into python extension (else there's too much indirection. If jupyter is knows about what version is required and checking the version to enable something, why move that logic into another extension, also makey shipping the two difficult, i.e. if we ship this change, we MUST have python change, else things will never work)

Reply:
`isProductVersionCompatible` just means whether the installed version satisfies a given semver constraint (point taken that the name could be improved). We can check if `isProductVersionCompatible` with the min version required for the new ipykernel behavior i.e. 6.0.x. If `false` that means use the old way, if `true` then we go down the codepath you added here. 

The reason I'm suggesting reusing `isProductVersionCompatible` is so that we don't duplicate logic for determining the version of `ipykernel` in the Jupyter extension.

Reply:
Oh yes, we can do that.